src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1323,50 +1329,53 @@ protected override void InvokeDefaultAction(string path)
 
             string resource = StringUtil.Format(FileSystemProviderStrings.InvokeItemResourceFileTemplate, path);
 
+            void StartProcessUsingHandler(string filename)
+            {
+                System.Diagnostics.Process process = new System.Diagnostics.Process();
+#if UNIX
+                // The file is possibly not an executable, so we try invoking the default program that handles this file.
+                const string quoteFormat = "\"{0}\"";
+                process.StartInfo.FileName = Platform.IsLinux ? "xdg-open" : /* OS X */ "open";
+                if (NativeCommandParameterBinder.NeedQuotes(filename))
+                {
+                    path = string.Format(CultureInfo.InvariantCulture, quoteFormat, filename);
+                }
+                process.StartInfo.Arguments = filename;
+                // xdg-open on Ubuntu outputs debug info to stderr, suppress it
+                process.StartInfo.RedirectStandardError = true;
+                process.Start();
+#else
+                if (Platform.IsNanoServer || Platform.IsIoT) { throw new NotSupportedException(); }
+                process.StartInfo.FileName = filename;
+                ShellExecuteHelper.Start(process.StartInfo);
+#endif
```

Comment:
So we are not differentiating FullCLR and CoreCLR on Windows. Are we supposed to ignore the FullCLR condition from now on? (which I'm totally fine with :))